### PR TITLE
1792 Improve web view experience for privacy policy

### PR DESCRIPTION
**Pull request by zamorastp**

### CODE REVIEW

File: Sources/AccountSDKIOSWeb/Simplified Login/SimplifiedLoginUIFactory.swift [hunk 1]
```diff
@@ -50,16 +50,19 @@ public struct SimplifiedLoginUIFactory {
...
+        let webVC = WebViewController()
+        webVC.loadURL(url)
+        
         viewModel.onClickedContinueAsUser = {} // TODO:
         
         viewModel.onClickedContinueWithoutLogin = {
             nc.dismiss(animated: true, completion: nil)
         }
         
         viewModel.onClickedPrivacyPolicy = { // TODO: Connect so that the text is a button or something else actionable
```
Comment (Daniel-Echegaray) [inline Sources/AccountSDKIOSWeb/Simplified Login/SimplifiedLoginUIFactory.swift:63]: This todo could be removed.

File: Sources/AccountSDKIOSWeb/Simplified Login/WebViewController.swift [hunk 2]
```diff
@@ -1,19 +1,91 @@
...
-        super.init(nibName: nil, bundle: nil)
+    func loadURL(_ url: URL) {
+        DispatchQueue.main.async {
+            self.webView.load(URLRequest(url: url))
+        }
+    }
+    
+    private func addLoadingViews() {
+        guard let navigationBar = navigationController?.navigationBar else {
+            return
```
Comment (Daniel-Echegaray) [inline Sources/AccountSDKIOSWeb/Simplified Login/WebViewController.swift:37]: Will ths mean If no navigation controller exists. Then no loading indicator will be shown, not even the spinner?

File: Sources/AccountSDKIOSWeb/Simplified Login/WebViewController.swift [hunk 3]
```diff
@@ -1,19 +1,91 @@
...
+
```
Comment (Daniel-Echegaray) [inline Sources/AccountSDKIOSWeb/Simplified Login/WebViewController.swift:38]: Extra blank space

File: Sources/AccountSDKIOSWeb/Simplified Login/SimplifiedLoginUIFactory.swift [hunk 4]
```diff
@@ -50,16 +50,19 @@ public struct SimplifiedLoginUIFactory {
         let nc = UINavigationController()
         nc.pushViewController(s, animated: false)
         
+        let url = URL(string: viewModel.localisation.privacyPolicyURL)!
+        let webVC = WebViewController()
+        webVC.loadURL(url)
```
Comment (Daniel-Echegaray) [inline Sources/AccountSDKIOSWeb/Simplified Login/SimplifiedLoginUIFactory.swift:55]: Line is 64 already calls loadURL. Is both of them needed?

**Review by Daniel-Echegaray — COMMENTED** [review]
Looks good 👌 
But I added a few questions.

File: Sources/AccountSDKIOSWeb/Simplified Login/WebViewController.swift (hunk 2 above)
Comment (zamorastp) [reply Sources/AccountSDKIOSWeb/Simplified Login/WebViewController.swift:37]: There will be a spinner without progress bar :)

File: Sources/AccountSDKIOSWeb/Simplified Login/SimplifiedLoginUIFactory.swift (hunk 1 above)
Comment (zamorastp) [reply Sources/AccountSDKIOSWeb/Simplified Login/SimplifiedLoginUIFactory.swift:63]: Sure, will do that

File: Sources/AccountSDKIOSWeb/Simplified Login/SimplifiedLoginUIFactory.swift (hunk 4 above)
Comment (zamorastp) [reply Sources/AccountSDKIOSWeb/Simplified Login/SimplifiedLoginUIFactory.swift:55]: Yes and this is why. User:
1.  opens privacy policy and browses through the pages 
2.  goes back to simplified login main view 
3. opens privacy policy again (should be privacy policy first page - not the last which he saw)

WkWebView caches content so there is no performance issues

File: Sources/AccountSDKIOSWeb/Simplified Login/WebViewController.swift (hunk 2 above)
Comment (Daniel-Echegaray) [reply Sources/AccountSDKIOSWeb/Simplified Login/WebViewController.swift:37]: This guard will require `WebViewController` to be embedded inside a navigation controller, to run the rest of the function logic. 

If that is the intention for the progress bar and not the spinner then please modify the guard to cover only the progress bar logic.

Comment (zamorastp) [reply Sources/AccountSDKIOSWeb/Simplified Login/WebViewController.swift:37]: Oh yes, i missed it 🤦‍♀️ Thank you 🙌

File: Sources/AccountSDKIOSWeb/Simplified Login/SimplifiedLoginUIFactory.swift (hunk 4 above)
Comment (Daniel-Echegaray) [reply Sources/AccountSDKIOSWeb/Simplified Login/SimplifiedLoginUIFactory.swift:55]: Ok I see. But if we only keep loadURL in line 64? Then wont this give us the same behaviour as you mention? (i.e. first page always opening) :)
Or maybe I am missing something 🤔

Comment (zamorastp) [reply Sources/AccountSDKIOSWeb/Simplified Login/SimplifiedLoginUIFactory.swift:55]: I think that now we gain on loading it previously 🤔

Comment (Daniel-Echegaray) [reply Sources/AccountSDKIOSWeb/Simplified Login/SimplifiedLoginUIFactory.swift:55]: What is the gain. Is it a performance gain?

Comment (zamorastp) [reply Sources/AccountSDKIOSWeb/Simplified Login/SimplifiedLoginUIFactory.swift:55]: If we are loading it earlier, should be faster I think

Comment (zamorastp) [reply Sources/AccountSDKIOSWeb/Simplified Login/SimplifiedLoginUIFactory.swift:55]: Yes, there is a little faster for the first time

Comment (zamorastp) [reply Sources/AccountSDKIOSWeb/Simplified Login/SimplifiedLoginUIFactory.swift]: I removed the first loadURL call :)

Comment (Daniel-Echegaray) [reply Sources/AccountSDKIOSWeb/Simplified Login/SimplifiedLoginUIFactory.swift]: Do you know if the WKWebView caches?

Comment (Daniel-Echegaray) [reply Sources/AccountSDKIOSWeb/Simplified Login/SimplifiedLoginUIFactory.swift]: If so then it might be good to have the first loadUrl?

Comment (zamorastp) [reply Sources/AccountSDKIOSWeb/Simplified Login/SimplifiedLoginUIFactory.swift]: let's leave it as it is and loadURL only on demand :) sounds more natural

Comment (zamorastp) [reply Sources/AccountSDKIOSWeb/Simplified Login/SimplifiedLoginUIFactory.swift]: still looks pretty good

Comment (Daniel-Echegaray) [reply Sources/AccountSDKIOSWeb/Simplified Login/SimplifiedLoginUIFactory.swift]: Ok 👍

**Review by Daniel-Echegaray — APPROVED** [review]

**Pull request merged**